apps/emqx_rule_engine/src/emqx_rule_actions.erl

```diff
@@ -219,18 +226,25 @@ on_action_republish(Selected, _Envs = #{
             metadata := Metadata}) ->
     ?LOG_RULE_ACTION(debug, Metadata, "[republish] republish to: ~p, Selected: ~p", [TargetTopic, Selected]),
     TargetRetain = maps:get('TargetRetain', Bindings, false),
-    Message =
-        #message{
-            id = emqx_guid:gen(),
-            qos = get_qos(TargetQoS, Selected, 0),
-            from = ActId,
-            flags = #{dup => false, retain => get_retain(TargetRetain, Selected)},
-            headers = #{republish_by => ActId},
-            topic = emqx_rule_utils:proc_tmpl(TopicTks, Selected),
-            payload = format_msg(PayloadTks, Selected),
-            timestamp = erlang:system_time(millisecond)
-        },
-    increase_and_publish(ActId, Message).
+    Topic = emqx_rule_utils:proc_tmpl(TopicTks, Selected),
+    case emqx_topic:wildcard(Topic) of
```

Comment:
If we do assertion the topic here, the action will be treated as 'failed', and rule-engine will increase the error counter for you automatically.

```suggestion
    assert_topic_valid(Topic)
```